Update README.md (#39)
mathworks-ref-arch/matlab-production-server-on-kubernetes: README.md

For more information about Kubernetes, see the [Kubernetes documentation](https://kubernetes.io/docs/home/).

## Contents
- [Requirements](#requirements)
- [Quick Start](#quick-start)
- [Deployment Steps](#deployment-steps)
  1. [Clone GitHub Repository](#step-1-clone-github-repository-that-contains-helm-chart)
  2. [Pull Container Images](#step-2-pull-container-images-for-matlab-production-server-and-matlab-runtime)

>   3. [Upload Container Images to Private Registry](#step-3-upload-container-images-to-private-registry-optional) *(optional)*

Reviewer: ▎ The anchor includes -optional but the actual heading (### Step 3: Upload Container Images to Private Registry) won't generate 
  ▎ that suffix — this link will be broken.

  ```suggestion
    3. [Upload Container Images to Private Registry](#step-3-upload-container-images-to-private-registry) *(optional)*
  ```

* You don't require significant changes to the Helm chart.
* For CI/CD workflows, we recommend that you retag and cache docker images in your private container registry.

> The Quick Start option only requires you to download a single file, rather than cloning the full GitHub repository. For more complex workflows, use the [Deployment Steps](#Deployment-Steps).

Reviewer: ▎ All other anchors in this PR use lowercase (#deployment-steps, #quick-start, etc.). This one still uses title-case. GitHub 
  ▎ anchors are case-insensitive so it works, but normalizing avoids confusion.

  ```suggestion
  The Quick Start option only requires you to download a single file, rather than cloning the full GitHub repository. For more
  complex workflows, use the [Deployment Steps](#deployment-steps).
  ```

    ```
    kubectl scale deployment matlab-production-server --namespace=<k8s-namespace> --replicas=4
    ```
* **Number of workers per pod** &mdash; Set the `--num-workers` option in the `values-overrides.yaml` file under `matlabProductionServerSettings` to control how many MATLAB workers run inside each pod.

#### Best Practices and Restrictions

* **License limits** &mdash; The total number of workers across all pods cannot exceed your MATLAB Production Server license seat count. For example, if you have a 4-seat license and configure 2 workers per pod, you can run at most 2 pods.
* **Resource requirements** &mdash; As of R2025a, the default configuration creates 2 workers per pod, with CPU and memory requirements based on existing product recommendations. This results in a resource request of approximately 1 CPU per pod. When scaling the number of pods, ensure your Kubernetes cluster has sufficient resources to accommodate the total CPU and memory requirements, as it is easy to exceed overall cluster resource limits.

> * **Choosing between more pods vs. more workers** &mdash; More pods provides better fault isolation and allows Kubernetes to distribute load across nodes. More workers per pod reduces scheduling overhead but increases per-pod resource requirements.

Reviewer: ▎ Subject-verb agreement: "More pods" is plural, so it should be "provide" and "allows" → "allow".

  ```suggestion
  * **Choosing between more pods vs. more workers** &mdash; More pods provide better fault isolation and allow Kubernetes to distribute
  load across nodes. More workers per pod reduces scheduling overhead but increases per-pod resource requirements.
  ```

To restart a single pod:
```
kubectl delete pod <podname> --namespace=<k8s-namespace>
```

To restart all pods, scale the deployment to 0 and then back to the desired number:
```
kubectl scale deployment matlab-production-server --namespace=<k8s-namespace> --replicas=0
kubectl scale deployment matlab-production-server --namespace=<k8s-namespace> --replicas=<desired-count>
```

### Common Issues

#### License Errors
* MATLAB Production Server on Kubernetes requires a **concurrent** license.
* The license server must be reachable from the network within the Kubernetes cluster but must not be installed in the cluster.
* If the server has difficulty resolving the DNS for a license server specified by hostname, try using the license server's IP address instead (e.g., `27000@172.22.225.0` instead of `27000@MYLICENSEHOST`).
* Each pod requires enough license seats for all its workers. For example, with the default of 2 workers per pod, you need at least 2 license seats per pod.

For more information, see [How can I troubleshoot license errors when using MATLAB Production Server on Kubernetes?](https://www.mathworks.com/matlabcentral/answers/672216)

#### Container Download Issues

> If you encounter issues downloading container images, see [Why am I encountering issues downloading containers for my MATLAB Production Server Kubernetes deployment?](https://www.mathworks.com/matlabcentral/answers/659239)

Reviewer: answers/659239 page does not exist
